### PR TITLE
Dynamic Vegetation automated test conversion/optimization

### PR DESCRIPTION
**Pull request by jckand-amzn**

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/EmptyInstanceSpawner_EmptySpawnerWorks.py [hunk 1]
```diff
@@ -5,109 +5,131 @@
...
-            heightmap_resolution=1024,
-            heightmap_meters_per_pixel=1,
-            terrain_texture_resolution=4096,
-            use_terrain=False,
+
+class BehaviorContextTests:
+    spawner_initialized = (
+        "Successfully initialized a Dynamic Slice Instance Spawner",
+        "Failed to initialize a Dynamic Slice Instance Spawner"
+    )
```
Comment (mbalfour-amzn) [inline AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/EmptyInstanceSpawner_EmptySpawnerWorks.py:13]: Shouldn't these say "Empty Instance Spawner"?

File: AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/MeshBlocker_InstancesBlockedByMeshHeightTuning.py [hunk 2]
```diff
@@ -5,104 +5,100 @@
...
+    """
+    Summary:
+    A temporary level is created, then a simple vegetation area is created. A blocker area is created and it is
+    verified that the tuning of the height percent blocker setting works as expected.
+
+    Expected Behavior:
+    Vegetation is blocked only around the trunk of the tree, while it still plants under the areas covered by branches.
+
+    Test Steps:
+     1) Opem a level
```
Comment (mbalfour-amzn) [inline AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/MeshBlocker_InstancesBlockedByMeshHeightTuning.py:26]: "Opem" -> "Open"

File: AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/ShapeIntersectionFilter_FilterStageToggle.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    # Open an existing simple level
+    helper.init_idle()
+    helper.open_level("Physics", "Base")
+
+    general.set_current_view_position(512.0, 480.0, 38.0)
+
+    # Create basic vegetation entity
+    position = math.Vector3(512.0, 512.0, 32.0)
+    asset_path = os.path.join("Slices", "PinkFlower.dynamicslice")
+    vegetation = dynveg.create_vegetation_area("vegetation", position, 16.0, 16.0, 16.0, asset_path)
```
Comment (mbalfour-amzn) [inline AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/ShapeIntersectionFilter_FilterStageToggle.py:60]: Just a random note - we should probably start using prefabs for any new tests written.  I believe the code can successfully spawn both dynamic slices _and_ prefabs regardless of whether or not prefabs are enabled for the project.

Comment (jckand-amzn) [reply AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/ShapeIntersectionFilter_FilterStageToggle.py:60]: Agreed. This test isn't actually new, it was misnamed previously so went through a rename, but will definitely keep that in mind for new tests, and plan to update all of these to use prefabs in the near future.

File: AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/EmptyInstanceSpawner_EmptySpawnerWorks.py (hunk 1 above)
Comment (jckand-amzn) [reply AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/EmptyInstanceSpawner_EmptySpawnerWorks.py]: Fixed! Good catch!

File: AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/MeshBlocker_InstancesBlockedByMeshHeightTuning.py (hunk 2 above)
Comment (jckand-amzn) [reply AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/MeshBlocker_InstancesBlockedByMeshHeightTuning.py]: Fixed!

**Review by mbalfour-amzn — APPROVED** [review]

**Pull request merged**